sdv/vdb/client.py

```diff
@@ -41,16 +40,20 @@ class VehicleDataBrokerClient:
     def __new__(cls, port: Optional[int] = None):
         if cls._instance is None:
             cls._instance = super(VehicleDataBrokerClient, cls).__new__(cls)
-            if not conf.DISABLE_DAPR:
-                if port is None:
-                    port = int(str(os.getenv("DAPR_GRPC_PORT")))
-                cls._address = f"localhost:{port}"
+            service_locator = config.middleware.service_locator
+            _location = service_locator.get_service_location("vehicledatabroker")
+            _hostname = urlparse(_location).hostname
+            if port is None:
+                _port = urlparse(_location).port
             else:
-                cls._address = conf.VEHICLE_DATA_BROKER_ADDRESS
+                _port = port
```

Comment:
I seems a bit strange, that possible content of `port` parameter gets precedence over a possible port set in the `_location`.

Reply:
The port can be overwritten by the developer and may replace the one from the service locator.